### PR TITLE
[core] Flip implied cxx abi for mason packages

### PR DESCRIPTION
**Pull request by yhahn**
Switches to new behavior:

- No suffix means `cxx11abi` in `mason`
- Explicit `cxx03abi` suffix required

cc @jfirebaugh @tmpsantos

### CODE REVIEW

File: scripts/check-cxx11abi.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ fi
 LIBSTDCPP=$(ldd $(dirname $0)/check-cxx11abi.dat |grep libstdc++ |cut -d' ' -f3)
```
Comment (tmpsantos) [inline scripts/check-cxx11abi.sh]: I think now that we inverted the semantics of this script, we should rename it to `check-cxx03abi`

File: platform/qt/scripts/configure.sh [hunk 2]
```diff
@@ -8,7 +8,7 @@ BOOST_VERSION=1.60.0
 GEOMETRY_VERSION=0.8.0
 GEOJSON_VERSION=0.1.4${CXX11ABI:-}
 GEOJSONVT_VERSION=6.1.0
-GTEST_VERSION=1.7.0${CXX11ABI:-}
+GTEST_VERSION=1.7.0-1${CXX11ABI:-}
```
Comment (tmpsantos) [inline platform/qt/scripts/configure.sh]: What about `CXX11ABI` -> `CXX03ABI`?

**Comment by tmpsantos** [comment]
The binary you uploaded for this one is crashing.

**Comment by jfirebaugh** [comment]
Closing; we couldn't find a solution here and the work here went stale.

**Pull request closed without merging**